PyInstaller/hooks/hook-matplotlib.backends.py

```diff
@@ -8,8 +8,73 @@
 #-----------------------------------------------------------------------------
 
 
-from PyInstaller.utils.hooks.hookutils import matplotlib_backends
+from PyInstaller.compat import is_darwin
+from PyInstaller.utils.hooks.hookutils import (
+    eval_statement, exec_statement, logger)
 
 
-# Include only available matplotlib backends.
-hiddenimports = matplotlib_backends()
+def get_matplotlib_backend_module_names():
+    """
+    List the names of all matplotlib backend modules importable under the
+    current Python installation.
+
+    Returns
+    ----------
+    list
+        List of the fully-qualified names of all matplotlib backend modules.
+    """
+    backend_names = eval_statement(
+        'import matplotlib; print(matplotlib.rcsetup.all_backends)')
+    module_names = []
+    import_statement = """
+import os, sys
+
+# Preserve stdout.
+sys_stdout = sys.stdout
+
+try:
+    # Redirect output printed by this importation to "/dev/null", preventing
+    # this output from being erroneously interpreted as an error.
+    with open(os.devnull, 'w') as dev_null:
+        sys.stdout = dev_null
+        __import__('%s')
+except Exception as exc:
+    # Restore stdout.
+    sys.stdout = sys_stdout
+
+    # If this is an ImportError, print this exception's message without printing
+    # a traceback. (ImportError messages are usually human-readable and thus
+    # require no tracebacks for additional context.)
+    if isinstance(exc, ImportError):
```

Comment:
**Because...** that's why.

Very well! You're right. Less magic makes more happiness. I'll tidy this up.